nefertari_sqla/fields.py

```diff
@@ -95,6 +95,9 @@ def process_type_args(self, kwargs):
             * type_kw: dict of type-specific kwargs
             * cleaned_kw: input kwargs cleaned from type-specific args
         """
+        if 'custom_analyzer' in kwargs:
+            self._custom_analyzer = kwargs.get('custom_analyzer')
```

Comment:
You could simply use `kwargs['custom_analyzer']` since you already checked it was there.
Or get rid of the `if` and do:
```python
self._custom_analyzer = kwargs.get('custom_analyzer') or self._custom_analyzer
```